### PR TITLE
exp show: Alternate row background color

### LINKED ISSUE #7128: exp show: alternate background colors slightly to make navigation easier
It would be way easier to navigate in the table if we had different background for items nearby, e.g. VSCode:

Screen Shot 2021-12-10 at 7 12 51 PM

### PR DESCRIPTION
**Pull request by daavoo**
* [X] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Closes #7128 

Kind of tricky to get a visually pleasant style (related https://github.com/willmcgugan/rich/discussions/363)

Here is the current state:

Captura de pantalla 2021-12-14 a las 18 14 41

Captura de pantalla 2021-12-14 a las 18 12 55

### CODE REVIEW

**Review by pared — APPROVED** [review]

**Comment by dberenbaum** [comment]
Nice work, @daavoo! Have you tried dropping the borders? Wondering if it's helpful or distracting as is.

@shcheklein what do you think?

**Comment by shcheklein** [comment]
I like it, and box simple makes total sense to me. 

Also, I wonder if we need bold font (or is it your local settings, @daavoo ?)

**Comment by daavoo** [comment]
> Nice work, @daavoo! Have you tried dropping the borders? Wondering if it's helpful or distracting as is.
> 
> @shcheklein what do you think?

Updated description showing `box.SIMPLE` mode (removing borders).

**Comment by daavoo** [comment]
> Also, I wonder if we need bold font (or is it your local settings, @daavoo ?)

We set bold font for `baseline` experiments (no differences in the table I show which is using `--all-tags`). Here is how it looks different with no baseline experiments:

Captura de pantalla 2021-12-14 a las 18 03 50

File: dvc/command/experiments/show.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def prepare_exp_id(kwargs) -> "RichText":
 
 
 def baseline_styler(typ):
-    return {"style": "bold"} if typ == "baseline" else {}
+    return {"style": "bold"} if typ == "baseline" else {"style": "default"}
```
Comment (skshetry) [inline dvc/command/experiments/show.py:376]: hmm, I don't see a "default" style in https://github.com/willmcgugan/rich/blob/2ba277ac54a153d3030b496698133ec2e4f67e1d/rich/default_styles.py. Does this even work?

Comment (daavoo) [reply dvc/command/experiments/show.py:376]: It works but it is was only added to make it easier to set `on {bcolor}`

Comment (skshetry) [reply dvc/command/experiments/show.py:376]: @daavoo, can we remove it now?

Comment (daavoo) [reply dvc/command/experiments/show.py:376]: Yes. I Forgot to do it in #7212

**Pull request merged**